### PR TITLE
JWT parse failure and unit test fixes

### PR DESCRIPTION
**Pull request by Fitzpasd**
Found this issue when I was working on the XBL sign in flow.
- Fixed issue
- Added unit test for this flow
- Fixed other unit test (it was 'passing' by swallowing an exception and never evaluating the assertions)

### CODE REVIEW

File: src/main/java/pro/beam/api/http/BeamHttpClient.java [hunk 1]
```diff
@@ -1,13 +1,14 @@
 package pro.beam.api.http;
 
+import com.auth0.jwt.internal.org.bouncycastle.util.encoders.Base64;
```
Comment (JamyDev) [inline src/main/java/pro/beam/api/http/BeamHttpClient.java]: Why import this, doesn't Java already have a Base64 parser? 
https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html

File: src/test/java/pro/beam/api/test/unit/http/BeamHttpClientTest.java [hunk 2]
```diff
@@ -92,4 +101,62 @@ private HttpResponse prepareResponse(int expectedResponseStatus, String expected
         }
         return response;
     }
+
+    @Test
+    public void itHandlesJWT() {
```
Comment (JamyDev) [inline src/test/java/pro/beam/api/test/unit/http/BeamHttpClientTest.java:106]: Nice, have some 🍰 !

**Review by JamyDev — APPROVED** [review]
Minor comment

File: src/main/java/pro/beam/api/http/BeamHttpClient.java (hunk 1 above)
Comment (Fitzpasd) [reply src/main/java/pro/beam/api/http/BeamHttpClient.java]: I'll check this out, think this was an auto-import.

Comment (Fitzpasd) [inline src/main/java/pro/beam/api/http/BeamHttpClient.java]: Sadly that util isn't available in jdk 1.7

Comment (Fitzpasd) [reply src/main/java/pro/beam/api/http/BeamHttpClient.java]: Changed to a different JRE component (that one is only available in Java 8)

**Pull request merged**